File: app/src/main/java/com/mert/bilgram/RecyclerViewAdapter.java
package com.mert.bilgram;

import android.app.Activity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {

    private final ArrayList<String> userEmail;
    private final ArrayList<String> postDesc;
    private final ArrayList<String> userImage;
    private Activity context;
    private FeedActivity feedActivity;

    RecyclerViewAdapter(ArrayList<String> userEmail, ArrayList<String> postDesc, ArrayList<String> userImage, final Activity context, final FeedActivity feedActivity) {
        this.userEmail = userEmail;
        this.postDesc = postDesc;
        this.userImage = userImage;
        this.context = context;
        this.feedActivity = feedActivity;

    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {

        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.recycler_view, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder viewHolder, int i) {

        Glide.with(context).load(userImage.get(i)).into(viewHolder.imageView);
        viewHolder.userEmail.setText(userEmail.get(i));
        viewHolder.postDesc.setText(postDesc.get(i));

        viewHolder.parentLayout.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View view) {
                return feedActivity.itemLongClicked(viewHolder.getAdapterPosition());
            }
        });

    }

    @Override
    public int getItemCount() {
        Log.e("RecyclerViewAdapter", "getItemCount: " + userEmail.size());
        return userEmail.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView userEmail;
        ImageView imageView;
        TextView postDesc;
        ConstraintLayout parentLayout;

        ViewHolder(@NonNull View itemView) {
            super(itemView);

            userEmail = itemView.findViewById(R.id.userEmailTVCV);
            imageView = itemView.findViewById(R.id.imageViewCV);
            postDesc = itemView.findViewById(R.id.descTVCV);
            parentLayout = itemView.findViewById(R.id.parent_layout);

        }
    }

}

File: app/src/main/java/com/mert/bilgram/RecyclerViewAdapter.kt
package com.mert.bilgram

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import java.util.*

class RecyclerViewAdapter internal constructor(private val userEmail: ArrayList<String?>, private val postDesc: ArrayList<String?>, private val userImage: ArrayList<String?>, private val context: Activity, private val feedActivity: FeedActivity) : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(viewGroup.context).inflate(R.layout.recycler_view, viewGroup, false))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        Glide.with(context).load(userImage[i]).into(viewHolder.imageView)
        viewHolder.userEmail.text = userEmail[i]
        viewHolder.postDesc.text = postDesc[i]
        viewHolder.parentLayout.setOnLongClickListener { feedActivity.itemLongClicked(viewHolder.adapterPosition) }
    }

    override fun getItemCount(): Int {
        return userEmail.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var userEmail: TextView = itemView.findViewById(R.id.userEmailTVCV)
        var imageView: ImageView = itemView.findViewById(R.id.imageViewCV)
        var postDesc: TextView = itemView.findViewById(R.id.descTVCV)
        var parentLayout: ConstraintLayout = itemView.findViewById(R.id.parent_layout)
    }

}